bulkExport/glueScripts/export-script.py

```diff
@@ -91,11 +100,70 @@ def remove_composite_id(resource):
                            datetime.strptime(x["meta"]["lastUpdated"], "%Y-%m-%dT%H:%M:%S.%fZ") <= datetime_transaction_time
                           )
 
+print ('start filtering by group_id')
+def is_active_group_member(member, datetime_transaction_time):
+    if getattr(member, 'inactive', None) == True:
+        return False
+    member_period = getattr(member, 'period', None)
+    if member_period != None:
+        end_date = getattr(member_period, 'end', None)
+        if end_date != None and datetime.strptime(end_date, "%Y-%m-%dT%H:%M:%S.%fZ") > datetime_transaction_time:
+            return False
+    return True
+
+def is_internal_reference(reference, server_url):
+    if reference.startswith(server_url):
+        reference = removeprefix(reference, server_url)
+    reference_split = reference.split('/')
+    if len(reference_split) == 2:
+        return True
+    return False
+
+def deep_get(resource, path):
+    if resource is None:
+        return None
+    if len(path) is 1:
+        return resource[path[0]]['reference']
+    return deep_get(resource[path[0]], path.pop(0))
+
+def is_included_in_group_export(resource, group_member_ids, group_patient_ids, compartment_search_params, server_url):
+    # Check if resource is part of the group
+    if resource['id'] in group_member_ids:
+        return True
+    # Check if resource is part of the patient compartment
+    if resource['resourceType'] in compartment_search_params.keys():
```

Comment:
searching in a list of keys seems inefficient (`compartment_search_params.keys()`). It's better to just check if the key is on the dictionary directly:
```py
resource['resourceType'] in compartment_search_params
```